libarchive/archive_read_support_format_tar.c

```diff
@@ -129,7 +129,11 @@ struct tar {
 	int64_t			 entry_offset;
 	int64_t			 entry_padding;
 	int64_t 		 entry_bytes_unconsumed;
-	int64_t			 realsize;
+	int64_t			 disk_size;
+	int64_t			 GNU_sparse_realsize;
+	int64_t			 GNU_sparse_size;
+	int64_t			 SCHILY_sparse_realsize;
+	int64_t			 pax_size;
```

Comment:
Rather than a single `realsize` field, this now splits out separate fields for four different pax attributes we might see.  The new `disk_size` stores the final reconciled size of the file on disk.